subprojects/packagefiles/eyalroz-printf/test/meson.build

```diff
@@ -0,0 +1,75 @@
+common_compiler_args = [
+  '-g',  # gcc args
+  '-Wall',
+  '-Wextra',
+  '-pedantic',
+  '-Wundef',
+  '-Wsign-conversion',
+  '-Wuninitialized',
+  '-Wshadow',
+  '-Wunreachable-code',
+  '-Wswitch-default',
+  '-Wswitch',
+  '-Wcast-align',
+  '-Wmissing-include-dirs',
+  '-Winit-self',
+  '-Wdouble-promotion',
+  '-gdwarf-2',
+  '-fno-exceptions',
+  '-ffunction-sections',
+  '-fdata-sections',
+  '-fverbose-asm',
+  '-Wunused-parameter',
+  '-ffat-lto-objects',
+  '-Wstrict-prototypes',
+  '/W4',  # msvc args
+]
+tests_c_args = cc.get_supported_arguments(
+  common_compiler_args,
+  '-std=c11',
+  '/std:c11',
+)
+
+cxx = meson.get_compiler('cpp')
+tests_cpp_args = cxx.get_supported_arguments(
+  common_compiler_args,
+  '-std=c++11',
+  '/std:c++11',
+)
+
+aliasing_exe = executable(
+  'aliasing',
+  'aliasing.c',
+  cfg_header_file,
+  c_args: [cc.get_supported_arguments('-fno-builtin-printf'), tests_c_args],
+  dependencies: [eyalroz_printf_dep],
+  include_directories: '..',
```

Comment:
Use `build_by_default: false` for test executables. They don't need to be built when running `ninja && meson install`.